libdrgn/language_c.c

```diff
@@ -1700,7 +1700,7 @@ struct drgn_error *drgn_c_family_lexer_func(struct drgn_lexer *lexer,
 		if (isalpha(*p) || *p == '_') {
 			do {
 				p++;
-			} while (isalnum(*p) || *p == '_');
+			} while (isalnum(*p) || *p == '_' || (cpp && *p == ':'));
```

Comment:
Yes, I realize you'll probably hate this, but I did want to put it out there since it's ultimately a very simple way to solve the problem at hand.

Reply:
Ha, yeah, I'm still not a fan. But I think we can do something similar to my first suggestion for template arguments (https://github.com/osandov/drgn/commit/a2140d2da639d53daca671d6174e553d74e4f876): add a `C_TOKEN_COLON` token, then modify `cpp_append_template_arguments_to_identifier()` to skip over any sequence of `C_TOKEN_IDENTIFIER` or `C_TOKEN_COLON` (and probably rename `cpp_append_template_arguments_to_identifier()` to something more general like `cpp_append_to_identifier()`).